### ISSUE TITLE
torchscript runtime error

### ISSUE THREAD

**Issue by KnightOfTheMoonlight** [nerf2nerf/nerf2nerf#9]
Hi, lily, thanks for sharing this great work.

I tried to follow your guide for nerf wrapper in [issue](https://github.com/nerf2nerf/nerf2nerf/issues/6). 
I tried to run the demo:
```
python main.py --yaml bust
python distill.py --scene_no 2 --a_or_b a
```
.But I got the follow error about torchscript.

`RuntimeError: The following operation failed in the TorchScript interpreter.`

```
/home/sse/code/nerf2nerf/utils.py:130: UserWarning: FALLBACK path has been taken inside: runCudaFusionGroup. This is an indication that codegen Failed for some reason.
To debug try disable codegen fallback path via setting the env variable export PYTORCH_NVFUSER_DISABLE=fallback
 (Triggered internally at ../third_party/nvfuser/csrc/manager.cpp:335.)
  rgb_n1, density_n1 = nerf1.nerf.call_eval(positions_chunk1.to(device),
Traceback (most recent call last):
  File "main.py", line 183, in <module>
    uu.show(writer, R, t, wrapper1, wrapper2, s1, focal, X, As, time, scale=opt.scale,
  File "/home/sse/code/nerf2nerf/utils.py", line 215, in show
    rgb1 = nerf1.render_image_fine((s).unsqueeze(0).to(device).float(), focal_length=focal, scale=1, near=near, far=far, chunk=chunk)
  File "/home/sse/code/nerf2nerf/nerf_wrapper.py", line 124, in render_image_fine
    rgba_n, depth_n = self.nerf.trace_eval(  # coarse and fine sampling + render incorporated in nerf jit model
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript, serialized code (most recent call last):
  File "code/__torch__/model/nerf/___torch_mangle_75.py", line 102, in trace_eval
    _23 = torch.mul(torch.unsqueeze(_22, 1), torch.unsqueeze(depth0, 2))
    positions = torch.add(_21, _23)
    _24 = (coarse2).forward(positions, inputs, )
           ~~~~~~~~~~~~~~~~ <--- HERE
    _25, _26, = _24
    _27 = torch.slice(depth0, 1, 1, 9223372036854775807)
  File "code/__torch__/model/nerf.py", line 58, in forward
    density0 = torch.to(density, 6)
    _6 = torch.randn_like(density0, dtype=6, layout=0, device=torch.device("cuda:0"), pin_memory=False)
    density1 = torch.add_(density0, torch.mul(_6, CONSTANTS.c7))
               ~~~~~~~~~~ <--- HERE
    sample_density = torch.softplus(density1)
    _7 = (to_radiance).forward((radiance).forward(_5, ), )

Traceback of TorchScript, original code (most recent call last):
/home/drebain/Workspace/torch_nerf/model/nerf.py(474): query
/home/drebain/Workspace/torch_nerf/model/nerf.py(464): forward
/home/drebain/anaconda3/envs/torch_nerf/lib/python3.10/site-packages/torch/nn/modules/module.py(1092): _slow_forward
/home/drebain/anaconda3/envs/torch_nerf/lib/python3.10/site-packages/torch/nn/modules/module.py(1108): _call_impl
/home/drebain/Workspace/torch_nerf/model/nerf.py(234): trace_rays_importance
/home/drebain/Workspace/torch_nerf/model/nerf.py(512): trace_eval
/home/drebain/anaconda3/envs/torch_nerf/lib/python3.10/site-packages/torch/jit/_trace.py(939): trace_module
/home/drebain/Workspace/torch_nerf/serialize_model.py(42): main
/home/drebain/Workspace/torch_nerf/config.py(76): start_script
/home/drebain/Workspace/torch_nerf/serialize_model.py(8): <module>
RuntimeError: a view of a leaf Variable that requires grad is being used in an in-place operation.
```

Cause these is not source code part for this. I really have no clue for debugging this. Could you give any suggestions?

**Comment by lilygoli** [comment]
Hi,

I think this happens because of version inconsistency as reported in #1. Please try to run the code with the supported cuda and PyTorch versions. For the source of NeRF codes, a link to the code is included in #6. I cannot reproduce this error, but if you keep running into it a simple workaround is rewriting the trace_eval function to just call the query method, as that seems to still work fine with the newer versions of cuda and PyTorch.

**Comment by KnightOfTheMoonlight** [comment]
> ce).forward((radiance).forward(_5, ), )
> 
> Traceback of TorchScript, original code (most recent call last):
> /home/drebain/Workspace/torch_nerf/model/nerf.py(474): query
> /home/drebain/Workspace/torch_nerf/model/nerf.py(464): forward
> /home/drebain/anaconda3/envs/torch_nerf/lib/python3.10/site-packages/torch/nn/modules/module.py(1092): _slow_forward
> /home/drebain/anaconda3/envs/torch_nerf/lib/python3.10/site-packages/torch/nn/modules/module.py(1108): _call_impl
> /home/drebain/Workspace/torch_nerf/model/nerf.py(234): trace_rays_importance
> /home/drebain/Workspace/torch_nerf/model/nerf.py(512): trace_eval
> /home/drebain/anaconda3/envs/torch_nerf/lib/python3.10/site-packages/torch/jit/_trace.py(939): trace_module
> /home/drebain/Workspace/torch_nerf/serialize_model.py(42): main
> /home/drebain/Workspace/torch_nerf/config.py(76): start_script
> /home/drebain/Workspace/torch_nerf/serialize_model.py(8): <module>
> RuntimeError: a view of a leaf Variable that requires grad is being used in an in-place operation.

Yes. The errors looks quite similar as [1](https://github.com/nerf2nerf/nerf2nerf/issues/1).
I have checked my environment setting. 
```
python 3.10
pytorch2.0.1+cu117
```
I have changed the pytorch version to `1.12.1+cu113`, it's working now!
